ament_package/group_dependency.py

```diff
@@ -56,5 +56,7 @@ def evaluate_condition(self, context):
     def extract_group_members(self, packages):
         self.members = set()
         for pkg in packages:
-            if self.name in [g.name for g in pkg.member_of_groups]:
+            for g in pkg.member_of_groups:
+                assert g.evaluate_condition is not None
+            if self.name in [g.name for g in pkg.member_of_groups if g.evaluate_condition]:
```

Comment:
If this is asserted above, does it need to be added to the list comprehension condition? Or is the assert used for debugging and will be removed before merging ?

Do you have an example of the bug that fails without and passes with this patch ?

Reply:
> If this is asserted above, does it need to be added to the list comprehension condition?

I don't understand the question. The `assert` statement ensures that the condition is not `None` which mean it must have been set to something else before. Usually by calling `evaluate_condition`. The list comprehension than only selects the group which `condition` is `True` and skips the ones which are `False`.

> Or is the assert used for debugging and will be removed before merging ?

No, the `assert` ensures that the API is being used correctly. `evaluate_condition` must have been invoked on all the groups before calling `extract_group_members`.

> Do you have an example of the bug that fails without and passes with this patch ?

If you have a group tag with a condition and other packages being a member of that group without the patch *all* members are being reported as dependencies even thought the condition of the group evaluates to false.

Reply:
So what you mean is that the assert is meant to ensure that the function has been called and that you want to check for the result of the evaluation is True. So you want to check the evaluate**d**_condition and not evalute_condition right ?

Reply:
oh weird I didn't receive any of these notifications (didnt refresh for a few hours). Yeah makes more sense now. LGTM too